sql/src/main/java/io/crate/analyze/ConstraintsValidator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import io.crate.analyze.symbol.Reference;
+import io.crate.metadata.ReferenceInfo;
+import io.crate.metadata.doc.DocTableInfo;
+
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+
+public final class ConstraintsValidator {
+
+    public static void validate(Object value, ReferenceInfo referenceInfo) {
+        // Validate NOT NULL constraint
+        if (value == null && !referenceInfo.isNullable()) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                "Cannot insert null value for column %s", referenceInfo.ident().columnIdent().fqn()));
+        }
+    }
+
+    public static void validate(Object[] values, Reference[] references, DocTableInfo tableInfo) {
```

Comment:
Instead of depending on DocTableInfo could this depend on `Iterable<? extends ReferenceInfo>` only? Would make unit testing easier.